cvat/apps/engine/cloud_provider.py

```diff
@@ -31,10 +32,43 @@
 from cvat.apps.engine.models import CloudProviderChoice, CredentialsTypeChoice
 from cvat.apps.engine.utils import get_cpu_number
 
+class NamedBytesIO(BytesIO):
+    @property
+    def filename(self) -> Optional[str]:
+        return getattr(self, '_filename', None)
+
+    @filename.setter
+    def filename(self, value: str) -> None:
+        self._filename = value
+
 slogger = ServerLogManager(__name__)
 
 ImageFile.LOAD_TRUNCATED_IMAGES = True
 
+CPU_NUMBER = get_cpu_number()
+
+def normalize_threads_number(
+    threads_number: Optional[int], number_of_files: int
+) -> int:
+    threads_number = (
+        min(
+            CPU_NUMBER,
+            settings.CVAT_MAX_THREADS_NUMBER_FOR_DATA_DOWNLOADING,
```

Comment:
```suggestion
            settings.CVAT_MAX_THREADS_NUMBER_FOR_DATA_DOWNLOADING,
```
Consider renaming to `CVAT_MAX_DATA_DOWNLOADING_THREADS` for brevity.